src/network/connection.h

```diff
@@ -724,7 +712,7 @@ struct ConnectionEvent
 		return "Invalid ConnectionEvent";
 	}
 
-	void dataReceived(session_t peer_id_, const SharedBuffer<u8> &data_)
+	void dataReceived(session_t peer_id_, const Buffer<u8> &data_)
```

Comment:
i'm not sure that buffer is consistent on multiple thread access comapred to SharedBuffer

Reply:
`SharedBuffer` is not thread-safe, neither is `Buffer` (which btw the structure already contains) so nothing changes in that regard.
Access to queues/lists containing these is only done with locks regardless.